data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/parser/LogstashVisitor.java

```diff
@@ -0,0 +1,170 @@
+package org.opensearch.dataprepper.logstash.parser;
+
+import org.opensearch.dataprepper.logstash.exception.LogstashParsingException;
+import org.opensearch.dataprepper.logstash.LogstashBaseVisitor;
+import org.opensearch.dataprepper.logstash.LogstashParser;
+import org.opensearch.dataprepper.logstash.model.LogstashConfiguration;
+import org.opensearch.dataprepper.logstash.model.LogstashPlugin;
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+import org.opensearch.dataprepper.logstash.model.LogstashAttributeValue;
+import org.opensearch.dataprepper.logstash.model.LogstashPluginType;
+import org.opensearch.dataprepper.logstash.model.LogstashValueType;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.LinkedList;
+import java.util.ArrayList;
+
+/**
+ * Class to populate Logstash configuration model objects using ANTLR
```

Comment:
Can you add the following to the documentation? I am not familiar with ANTLR and how this whole process works. It took me a little while to understand where the `LogstashBaseVisitor` is coming from. I didn't realize I had to build my code to generate the class.
- What is a LogstashBaseVisitor? 
- How is a LogstashBaseVisitor generated?
- Where do all the methods come from?

Reply:
While this is confusing in a PR, this is all part of the build process and I think detailed documentation here is not quite appropriate. If anything, I'd keep it as simple as "... using ANTLR library classes and generated code."

Reply:
That works. Anything to point the reader in the right direction.